### DISCUSSION TITLE
How to Properly Add Description/BearerFormat to Security Scheme Objects in OpenAPI?

### DISCUSSION THREAD

**Discussion by stevenj** [fastapi/fastapi#9009] [Questions] [answered]
### First check

* [X] I added a very descriptive title to this issue.
* [X] I used the GitHub search to find a similar issue and didn't find it.
* [X] I searched the FastAPI documentation, with the integrated search.
* [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [X] I already read and followed all the tutorial in the docs and didn't find an answer.
* [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [X] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

Open API defines the following fields for Security Scheme Objects:
See: https://spec.openapis.org/oas/v3.0.2#security-scheme-object

If i define a subclass of HTTPBearer like so:
```python
class HTTPAppBearerToken(HTTPBearer):
    """HTTP Bearer Authorization the a App Token."""

    def __init__(self, *args, **kwargs):
        """Customize the init, so we can add docs."""
        super().__init__(bearerFormat="AppJWT", *args, **kwargs)
        self.model.description = "My Custom Bearer Token!"
```

I can see the description of the security object being generated.  But the bearerFormat isn't making it to the schema.  This also feels like the wrong way to set the description, but if i try and call: `super().__init__(bearerFormat="AppJWT", description="Something", *args, **kwargs)` python objects.

So i have two questions.
1. Am i defining bearerFormat wrong or is it seimply not finding its way into the schema?  I see it set properly in the model if i break just after the super() call.  
2. Am i defining the models description the correct way?

I couldn't find any documentation on setting these documentary fields for Security Scheme Objects.

**Comment by YuriiMotov** [comment, marked as answer]
```
HTTPBearer(
    *,
    bearerFormat=None,
    scheme_name=None,
    description=None,
    auto_error=True
)
```

https://fastapi.tiangolo.com/reference/security/#fastapi.security.HTTPBearer
